File: app/src/main/java/com/iseasoft/iseaiptv/listeners/VideoPlayerListener.java
package com.iseasoft.iseaiptv.listeners;

import com.iseasoft.iseaiptv.ui.fragment.PlayerType;
import com.iseasoft.iseaiptv.ui.fragment.VideoPlayerView;

public interface VideoPlayerListener {

    void onChangeStatus(VideoPlayerView playerView, @PlayerType.PlayerStatusType int type);

    void onError(VideoPlayerView playerView, @PlayerType.PlayerStatusType int error);

    void playbackProgress(VideoPlayerView playerView, int progress);

    void playbackBufferProgress(VideoPlayerView playerView, int buffer);
}

File: app/src/main/java/com/iseasoft/iseaiptv/listeners/VideoPlayerListener.kt
package com.iseasoft.iseaiptv.listeners

import com.iseasoft.iseaiptv.ui.fragment.PlayerType
import com.iseasoft.iseaiptv.ui.fragment.VideoPlayerView

interface VideoPlayerListener {

    fun onChangeStatus(playerView: VideoPlayerView, @PlayerType.PlayerStatusType type: Int)

    fun onError(playerView: VideoPlayerView, @PlayerType.PlayerStatusType error: Int)

    fun playbackProgress(playerView: VideoPlayerView, progress: Int)

    fun playbackBufferProgress(playerView: VideoPlayerView, buffer: Int)
}
